pkg/nfd-client/topology-updater/nfd-topology-updater.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topologyupdater
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	v1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+	"golang.org/x/net/context"
+	"sigs.k8s.io/node-feature-discovery/pkg/apihelper"
+	nfdclient "sigs.k8s.io/node-feature-discovery/pkg/nfd-client"
+	"sigs.k8s.io/node-feature-discovery/pkg/podres"
+	"sigs.k8s.io/node-feature-discovery/pkg/resourcemonitor"
+	pb "sigs.k8s.io/node-feature-discovery/pkg/topologyupdater"
+	"sigs.k8s.io/node-feature-discovery/pkg/utils"
+	"sigs.k8s.io/node-feature-discovery/pkg/version"
+)
+
+// Command line arguments
+type Args struct {
+	nfdclient.Args
+	NoPublish      bool
+	Oneshot        bool
+	KubeConfigFile string
+}
+
+type NfdTopologyUpdater interface {
+	nfdclient.NfdClient
+	Update(v1alpha1.ZoneList) error
+}
+
+type staticNodeInfo struct {
+	args                Args
+	resourcemonitorArgs resourcemonitor.Args
+	tmPolicy            string
+}
+
+type nfdTopologyUpdater struct {
+	nfdclient.NfdBaseClient
+	nodeInfo  *staticNodeInfo
+	certWatch *utils.FsWatcher
+	client    pb.NodeTopologyClient
+	stop      chan struct{} // channel for signaling stop
+}
```

Comment:
As suggested in the repply to an earlier comment I'd move args out of `staticNodeInfo` as they're not node information

```suggestion
type staticNodeInfo struct {
	tmPolicy            string
}

type nfdTopologyUpdater struct {
	nfdclient.NfdBaseClient

	args                Args
	resourcemonitorArgs resourcemonitor.Args
	nodeInfo  *staticNodeInfo
	certWatch *utils.FsWatcher
	client    pb.NodeTopologyClient
	stop      chan struct{} // channel for signaling stop
}
```

Reply:
done